### PR TITLE
Add warnings about using non-default values for secure etcd comms

### LINKED ISSUE #2989: Add callouts regarding etcd cluster security
### Description

etcd isn't secure by default, and the docs can do more to call this out.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages

operations/deploy-sensu/secure-sensu/#secure-etcd-peer-communication
operations/deploy-sensu/cluster-sensu/
operations/deploy-sensu/use-federation/
operations/deploy-sensu/etcdreplicators/

### Expected behavior

Documentation makes very clear that explicit, non-default configuration is necessary to secure etcd communication in transit, as well as to authenticate clients and peers (i.e. mutual TLS/client certificate auth)

### Current behavior

Some documents (e.g. use-federation) suggest appropriate configuration, but in config file examples it's not explicit, and in 

### Corrections needed

Use callouts or other formatting to highlight existing content, add callouts in places where etcd cluster and client communication is discussed to highlight the importance of securing etcd.

### Additional context

Users who expose an unsecured etcd installation to an untrusted network are vulnerable to unauthorized manipulation of Sensu configuration via etcd client connections.

Out of the box, Sensu's embedded etcd listens for unencrypted communication on ports 2379 and 2380. In order to secure etcd, whether embedded in Sensu or deployed externally, the following must be taken into consideration:

* Providing key and certificate for `etcd-cert-file` and `etcd-key-file` to secure client communication, as well as the `etcd-peer-cert-file` and `etcd-peer-key-file` to secure cluster communication
* Configuring `etcd-advertise-client-urls`, `etcd-initial-cluster`, `etcd-client-urls`, `etcd-listen-client-urls`, `etcd-listen-peer-urls` with non-default values in order to make use of the above cert and key files
* Configuring `etcd-client-cert-auth` and `etcd-peer-client-cert-auth` to ensure that etcd only allows connections from clients and peers which present a valid, trusted certificate.

Without due consideration of the above, even an etcd installation which requires https connections may still allow any client to connect without providing a username or password (see https://github.com/sensu/sensu-docs/pull/2792 for draft documentation on requiring RBAC authentication in addition to certificate authentication).

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds warnings and more information about replacing backend config defaults with explicit, non-default values to secure etcd communication in these docs sections:
- https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/secure-sensu/#secure-etcd-peer-communication
- https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/cluster-sensu/#sensu-backend-configuration
- https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/cluster-sensu/#use-an-external-etcd-cluster
- https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/use-federation/#step-1-configure-backends-for-tls
- https://docs.sensu.io/sensu-go/latest/operations/deploy-sensu/etcdreplicators/#spec-attributes

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2989

### CODE REVIEW

File: content/sensu-go/6.2/operations/deploy-sensu/secure-sensu.md [hunk 1]
```diff
@@ -40,6 +45,14 @@ etcd-peer-client-cert-auth: "true"
 etcd-peer-trusted-ca-file: "/path/to/your/peer/ca/file"
 {{< /code >}}
 
+To properly secure etcd communication, replace the default parameter values in your backend store configuration with non-default versions of these certificates, keys, and URLs:
+
+ - A certificate and key for the `etcd-cert-file` and `etcd-key-file` to secure client communication
+ - A certificate and key for the `etcd-peer-cert-file` and `etcd-peer-key-file` to secure cluster communication
+ - Non-default values for `etcd-listen-client-urls`, `etcd-listen-peer-urls`, and `etcd-initial-advertise-client-urls`
+
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/operations/deploy-sensu/secure-sensu.md:54]: The [issue notes](https://github.com/sensu/sensu-docs/issues/2989) don't mention `etcd-trusted-ca-file` or `etcd-peer-trusted-ca-file` -- do those need to be listed here too?

**Comment by hillaryfraley** [comment]
@cwjohnston I don't think my additions here completely satisfy the [issue](https://github.com/sensu/sensu-docs/issues/2989). Specifically, I was not sure how to incorporate these points:
- Sensu's embedded etcd listens for unencrypted communication on ports 2379 and 2380
- Without due consideration of the above, even an etcd installation which requires https connections may still allow any client to connect without providing a username or password (see https://github.com/sensu/sensu-docs/pull/2792 for draft documentation on requiring RBAC authentication in addition to certificate authentication)

File: content/sensu-go/6.2/operations/deploy-sensu/secure-sensu.md (hunk 1 above)
Comment (cwjohnston) [reply content/sensu-go/6.2/operations/deploy-sensu/secure-sensu.md:54]: These flags are often used, but not strictly required for mutual TLS. Custom CA certificates can be added to the system trust store in a way that allows mutual TLS to work without providing values for these flags.

**Comment by cwjohnston** [comment]
> @cwjohnston I don't think my additions here completely satisfy the [issue](https://github.com/sensu/sensu-docs/issues/2989). Specifically, I was not sure how to incorporate these points:
> 
> * Sensu's embedded etcd listens for unencrypted communication on ports 2379 and 2380

I think the following look like good places to call this out:

observability-pipeline/observe-schedule/backend/#configuration
operations/deploy-sensu/datastore/#use-default-event-storage

> * Without due consideration of the above, even an etcd installation which requires https connections may still allow any client to connect without providing a username or password (see #2792 for draft documentation on requiring RBAC authentication in addition to certificate authentication)

I think we need to add `etcd-client-cert-auth: true` to the example config in operations/deploy-sensu/secure-sensu/#secure-etcd-peer-communication , then add a callout explaining that etcd does not require authentication by default, and therefore etcd-client-cert-auth and etcd-peer-client-cert-auth are an important part of securing sensu's etcd datastore against unauthorized access.

Once we have a product and/or documentation fix for configuring Sensu to follow etcd RBAC, we can explain any options that make sense for deployments using etcd's own access controls.

**Comment by hillaryfraley** [comment]
Thank you, @cwjohnston -- updated. Wdyt?

edit: forgot to say I can add a note in https://github.com/sensu/sensu-docs/pull/2792 or create a new issue to capture the updates we want to make based on how that PR winds up.

File: content/sensu-go/6.2/operations/deploy-sensu/datastore.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ For longer retention of observability event data, integrate Sensu with a time se
 
 By default, Sensu uses its embedded etcd database to store configuration and event data.
 This embedded database allows you to get started with Sensu without deploying a complete, scalable architecture.
+Sensu's embedded etcd listens for unencrypted communication on [ports][19] 2379 (client requests) and 2380 (peer communication).
```
Comment (cwjohnston) [inline content/sensu-go/6.2/operations/deploy-sensu/datastore.md]:
```suggestion
Sensu's default embedded etcd configuration listens for unencrypted communication on [ports][19] 2379 (client requests) and 2380 (peer communication).
```

**Review by cwjohnston — APPROVED** [review]

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/backend.md [hunk 3]
```diff
@@ -982,7 +982,7 @@ etcd-key-file: "./client-key.pem"{{< /code >}}
 
 | etcd-listen-client-urls |      |
 --------------------------|------
-description               | List of URLs to listen on for client traffic.
+description               | List of URLs to listen on for client traffic. Sensu's embedded etcd listens for unencrypted client communication on port 2379.
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/backend.md]:
```suggestion
description               | List of URLs to listen on for client traffic. Sensu's default embedded etcd configuration listens for unencrypted client communication on port 2379.
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/backend.md [hunk 4]
```diff
@@ -999,7 +999,7 @@ etcd-listen-client-urls:
 
 | etcd-listen-peer-urls |      |
 ------------------------|------
-description             | List of URLs to listen on for peer traffic.
+description             | List of URLs to listen on for peer traffic. Sensu's embedded etcd listens for unencrypted peer communication on port 2380.
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/backend.md]:
```suggestion
description             | List of URLs to listen on for peer traffic. Sensu's default embedded etcd configuration listens for unencrypted peer communication on port 2380.
```

**Pull request merged**